kong/router.lua

```diff
@@ -709,7 +710,8 @@ do
     end,
 
     [MATCH_RULES.URI] = function(category, ctx)
-      return category.routes_by_uris[ctx.hits.uri or ctx.req_uri]
+      return category.routes_by_uris[ctx.req_uri] or
+             category.routes_by_uris[ctx.hits.uri]
```

Comment:
This change can be detrimental in many cases (that is, instead of taking the hint into consideration, and prioritizing the evaluation of Routes with the hinted path, we now overlook the hint and perform the lookup based on the request path. The hint serves no purpose anymore.